File: VBTestSnippets/Classification.vb
' See "CSharpTestSnippets\Classification.cs" for many of the shared classification types between C# and VB
' This file only illustrates the classification types unique to VB

Module M ' "User Types - Modules"
    Sub M()
        Dim x = 7
        Dim y =
            <Container>
                <Test1/> ' "VB XML Literals - Delimiter" for the angle brackets
                <Test2/> ' "VB XML Literals - Name" for the "Test2"
                <Test3 Attribute=""/> ' "VB XML Literals - Attribute Name" for "Attribute"
                <Test4 Attribute="AttributeValue"/> ' "VB XML Literals - Attribute Value" for the "AttributeValue"
                <Test5 Attribute=""/> ' "VB XML Literals - Attribute Quotes" for the quotation marks
                <Text6>Text</Text6> ' "VB XML Literals - Text"
                <Test7><%= x %></Test7> ' "VB XML Literals - Embedded Expression" for the %=
                <Test8>&lt;</Test8> ' "VB XML Literals - Entity Reference"
                <Test9><?piName [= piData]?></Test9> "VB XML Literals - Processing Instruction"
                <Test10><![CDATA[content]]></Test10> "VB XML Literals - CData Section"
                <!-- Comment --> ' "VB XML Literals - Comment"
            </Container>
    End Sub
End Module

File: CSharpTestSnippets/Classification.cs
// This file illustrates the various classification types in VS

// "Comment"

using // "Keyword"
    System; // "Identifier"
using System.CodeDom; // Unused code fading (in Blue/Light/Dark themes)

namespace CSharpTestSnippets
{
    delegate int D(); // "User Types - Delegates"
    enum E { } // "User Types - Enums"
    interface I { } // "User Types - Interfaces"
    struct S { } // "User Types - Structures"
    class C1<TypeParameter> { } // "User Types - Type Parameters"

    class C2 // "User Types - Classes"
    {
        void M()
        {
            Console.WriteLine("Hello, world"); // "String"
            Console.WriteLine(@"Hello world"); // "String - Verbatim"

            var x1 = y; // Compiler error squiggle under "y"
            var x2 = 7; // Compiler warning squiggle under "x2"

            // Set "Tools | Options | Text Editor | C# | Code Style | General | 'var' preferences: | For built-in types" 
            // to "Prefer 'var'" with "Suggestion" severity
            int x3 = 7; // "Suggestion ellipses (...)" (under the "int")

            int x4 = 7 + 3; // "Operator" (the "+" sign)

#if FALSE
            int x4 = 7; // "Excluded Code"
#endif // "Preprocessor Keyword"

            Console.WriteLine("http://www.microsoft.com"); // "URL Hyperlink"
        }
    }

    // Right-click "Rename" on "ab" on the first line of "M" and change it to "abc"
    class InlineRenameSample
    {
        int abc = 7;

        void M()
        {
            int abc = 8; // "Inline Rename Field Text" + "Inline Rename Field Background and Border"
            this.abc = 8; // "Inline Rename Fixup"

            {
                int abc; // "Inline Rename Conflict
            }
        }
    }

    // Type a "1" at the end of "RenameTrackingSample" on the next line, you'll see a dotted border around it
    class RenameTrackingSample { } // "Rename Tracking"

    class DocCommentsSample
    {
        /// <summary> ("XML Doc Comments - Delimiter" for the "///", "XML Doc Comments - Name" for the "summary")
        /// Text ("XML Doc Comments - Text")
        /// <!--Comment--> ("XML Doc Comments - Comment")
        /// <![CDATA[ Content ]]> ("XML Doc Comments - CData Section")
        /// <tag Attribute = "Value"/>
        /// ("XML Doc Comments - Attribute Name" for "Attribute")
        /// ("XML Doc Comments - Attribute Quotes" for the quotation marks)
        /// ("XML Doc Comments - Attribute Value" for "Value")
        /// <tag><?test[=test]?></tag> ("XML Doc Comments - Processing Instruction")
        /// <tag>&lt;</tag> ("XML Doc Comments - Entity Reference" for "&lt;")
        /// </summary>
        void M() { }
    }
}
